### PR TITLE
[BACKLOG-1157] As a Sqoop Import user I need to select named clusters

### PR DESCRIPTION
**Pull request by e-cuellar**

### CODE REVIEW

**Comment by e-cuellar** [comment]
@mdamour1976 Please Sir!

File: src/org/pentaho/di/ui/vfs/VfsFileChooserHelper.java [hunk 1]
```diff
@@ -95,15 +108,15 @@ public FileObject browse( String[] fileFilters, String[] fileFilterNames, String
       selectedFile =
           fileChooserDialog
               .open( shell, this.schemeRestrictions, getDefaultScheme(), showFileScheme(), initialFile.getName()
-                  .getPath(), fileFilters, fileFilterNames, returnsUserAuthenticatedFileObjects(), fileDialogMode );
+                  .getPath(), fileFilters, fileFilterNames, returnsUserAuthenticatedFileObjects(), fileDialogMode, showLocation);
     } else {
       selectedFile =
           fileChooserDialog.open( shell, this.schemeRestrictions, getDefaultScheme(), showFileScheme(), null,
-              fileFilters, fileFilterNames, returnsUserAuthenticatedFileObjects(), fileDialogMode );
+              fileFilters, fileFilterNames, returnsUserAuthenticatedFileObjects(), fileDialogMode, showLocation);
```
Comment (mdamour1976) [inline src/org/pentaho/di/ui/vfs/VfsFileChooserHelper.java]: @e-cuellar formatting

File: src/org/pentaho/di/ui/job/AbstractJobEntryController.java [hunk 2]
```diff
@@ -330,14 +340,17 @@ protected FileObject browseVfs( FileObject root, FileObject initial, int dialogM
     fileChooserHelper.setDefaultScheme( defaultScheme );
     fileChooserHelper.setSchemeRestrictions( schemeRestrictions );
     fileChooserHelper.setShowFileScheme( showFileScheme );
+    if ( namedCluster != null ) {
+      fileChooserHelper.setNamedCluster( namedCluster );
+    }
     try {
-      return fileChooserHelper.browse( getFileFilters(), getFileFilterNames(), initial.getName().getURI(), dialogMode );
+      return fileChooserHelper.browse( getFileFilters(), getFileFilterNames(), initial.getName().getURI(), dialogMode, showLocation);
```
Comment (mdamour1976) [inline src/org/pentaho/di/ui/job/AbstractJobEntryController.java]: here

File: src/org/pentaho/di/ui/vfs/VfsFileChooserHelper.java [hunk 3]
```diff
@@ -61,10 +64,15 @@ public FileObject browse( String[] fileFilters, String[] fileFilterNames, String
...
     return browse( fileFilters, fileFilterNames, fileUri, VfsFileChooserDialog.VFS_DIALOG_OPEN_DIRECTORY );
   }
-
+  
   public FileObject browse( String[] fileFilters, String[] fileFilterNames, String fileUri, int fileDialogMode )
+      throws KettleException, FileSystemException {
+      return browse( fileFilters, fileFilterNames, fileUri, fileSystemOptions, fileDialogMode );
+  }
+
+  public FileObject browse( String[] fileFilters, String[] fileFilterNames, String fileUri, int fileDialogMode, boolean showLocation)
```
Comment (mdamour1976) [inline src/org/pentaho/di/ui/vfs/VfsFileChooserHelper.java]: here

File: src/org/pentaho/di/ui/vfs/VfsFileChooserHelper.java [hunk 4]
```diff
@@ -74,6 +82,11 @@ public FileObject browse( String[] fileFilters, String[] fileFilterNames, String
 
   public FileObject browse( String[] fileFilters, String[] fileFilterNames, String fileUri, FileSystemOptions opts,
       int fileDialogMode ) throws KettleException, FileSystemException {
+   return browse( fileFilters, fileFilterNames, fileUri, opts, fileDialogMode, true );
+  }
+  
+  public FileObject browse( String[] fileFilters, String[] fileFilterNames, String fileUri, FileSystemOptions opts,
+      int fileDialogMode, boolean showLocation) throws KettleException, FileSystemException {
```
Comment (mdamour1976) [inline src/org/pentaho/di/ui/vfs/VfsFileChooserHelper.java]: here

**Pull request merged**